boot/bootutil/src/bootutil_misc.c

```diff
@@ -17,7 +17,12 @@
  * under the License.
  */
 
+#ifdef __BOOTSIM__
```

Comment:
In general I don't like this sort of thing; we already have quite a bit of ifdeffery in mcuboot, and it makes it harder to audit/review. The fact that this definition has to happen in multiple files is also a sign of trouble.

Can you just build in the simulator with -Dassert=sim_assert or something like that? (Maybe adding an -include "place-where-sim_assert-is-declared.h" to the compile flags as well to get the declaration.)

Barring that, I think adding a header like bootutil_log.h for assertions, which abstracts the runtime environment in the same way, would be better -- especially since BOOT_STATUS_ASSERT() gets added later in the series.

Reply:
Hmm... and now I see even later in the series, the behavior changes even more.

Can you please clean this up a bit and squash the commits around so it's easier to review? The behavior of asserts seems to change a few times in this series, and it's not clear to me where this is going.